pytorch/training/docker/2.0/py3/cu121/Dockerfile.gpu

```diff
@@ -352,26 +353,40 @@ ARG PYTORCH_VERSION_EC2
 ARG CONDA_CHANNEL
 ARG CUDA_HOME
 ARG TORCH_CUDA_ARCH_LIST
+ARG SMD_MODEL_PARALLEL_CONDA_CHANNEL
 
 ENV SAGEMAKER_TRAINING_MODULE=sagemaker_pytorch_container.training:main
 
 WORKDIR /
 
-# Install ec2 PyTorch
-RUN /opt/conda/bin/mamba install -y python=${PYTHON_SHORT_VERSION} pytorch=${PYTORCH_VERSION_EC2} \
-  pytorch-cuda=12.1 cuda-nvcc=12.1.* torchvision torchaudio torchtext aws-ofi-nccl-dlc=1.7.3 \
+# Install aws-ofi-nccl, dependency of sm PyTorch
+RUN /opt/conda/bin/mamba install -y  \
+  "aws-ofi-nccl>=1.7.1,<2.0" \
   --override-channels \
-  -c ${CONDA_CHANNEL} \
+  -c https://aws-ml-conda.s3.us-west-2.amazonaws.com \
+  -c nvidia \
+  -c conda-forge \
+  && /opt/conda/bin/mamba clean -afy
+
+# Install sm PyTorch
+RUN /opt/conda/bin/mamba install -y python=${PYTHON_SHORT_VERSION} \
+  pytorch="2.0.1=sm_py3.10_cuda12.1*" \
+  pytorch-cuda=12.1 cuda-nvcc=12.1.* torchvision torchaudio torchtext \
+  --override-channels \
+  -c ${SMD_MODEL_PARALLEL_CONDA_CHANNEL} \
+  -c https://aws-ml-conda-ec2.s3.us-west-2.amazonaws.com \
+  -c pytorch \
+  -c numba/label/dev \
+  -c pytorch-nightly \
```

Comment:
why nightly is needed? does it have stable packages?

Reply:
for torchtrition which is a dependency for sm pytorch